client/browser/src/libs/bitbucket/context.tsx

```
@@ -0,0 +1,55 @@
+import { RepoSpec, RevSpec } from '../../../../../shared/src/util/url'
+import { CodeHostContext } from '../code_intelligence/code_intelligence'
+
+// example pathname: /projects/TEST/repos/testing/browse/src/extension.ts
+const PATH_REGEX = /^\/projects\/(\w+)\/repos\/(\w+)\//
+
+function getRepoSpecFromLocation(location: Pick<Location, 'hostname' | 'pathname'>): RepoSpec | null {
+    const { hostname, pathname } = location
+    const match = pathname.match(PATH_REGEX)
+    if (!hostname || !match) {
+        return null
```

We have a lot of functions in this PR that return `null` if they couldn't do what their function name says. Are these valid conditions, or are these error conditions? If the latter, they should throw a descriptive Error instead.

This function's `null` return value in particular seems to be converted to an exception by the caller anyway (but at a point where context is missing and the stack trace doesn't point to the real failure)